python/python/bystro/ancestry/train.py

```diff
@@ -308,6 +313,78 @@ def _calc_fst(variant_counts: pd.Series, samples: pd.DataFrame) -> float:
     return (p * (1 - p) - total) / (p * (1 - p))
 
 
+def load_1kgp_vcf(vcf_filepath: str) -> tuple[pd.DataFrame, pd.DataFrame]:
+    """Temporary placeholder method to load in 1kgp vcf
+    filtered down to the same variants as the gnomad loadings for testing
+    using plink2.
+    """
+    #This line will have to be altered if testing other vcfs
+    header_vcf=pd.read_csv(vcf_filepath,delimiter='\t',skiprows=107)
+    dosage_vcf = header_vcf.replace("0|0", 0)
+    dosage_vcf = dosage_vcf.replace("0|1", 1)
+    dosage_vcf = dosage_vcf.replace("1|0", 1)
+    dosage_vcf = dosage_vcf.replace("1|1", 2)
+    #If testing other vcfs this line will also need to be altered
+    dosage_vcf=dosage_vcf.rename(
+        columns={"#CHROM": "Chromosome", "POS": "Position"}, 
+        errors="raise"
+    )
+    genos = dosage_vcf.iloc[:, 9:]
+    genos["variant"] = dosage_vcf["ID"]
+    genos=genos.set_index("variant")
+    genos=genos.sort_index()
+    return dosage_vcf, genos
+    
+
+def load_pca_loadings(loadings: pd.DataFrame, dosage_vcf: pd.DataFrame) -> pd.DataFrame:
+    """Load in the gnomad PCs, reformats, and filters them to match
+    the format and variant list of the 1kgp vcf.
+    """
+    loadings[["Chromosome", "Position"]] = loadings["locus"].str.split(":", expand=True)
+    #Match variant format of gnomad loadings with 1kgp vcf
+    get_chr_pos= lambda x: x["locus"][3:]
+    get_allele1=  lambda x: x["alleles"][2]
+    get_allele2= lambda x: x["alleles"][6]
+    loadings["variant"] = loadings.apply(
+        lambda x: ":".join([get_chr_pos(x), get_allele1(x), get_allele2(x)]), axis=1
+    )
+    #Remove variants that are missing from IGSR version of 1kgp
+    missing_variants = set(loadings.variant) - set(dosage_vcf.ID)
+    loadings=loadings[~loadings["variant"].isin(missing_variants)]
+    #Remove brackets
+    loadings["loadings"] = loadings["loadings"].str[1:-1]
+    #Split PCs and join back
+    gnomadPCs = loadings["loadings"].str.split(",", expand=True)
+    loadings=loadings.join(gnomadPCs)
+    loadings=loadings.reset_index()
+    pc_range = range(8, 28)
+    pc_loadings = loadings.iloc[:, pc_range].copy()
+    pc_loadings["variant"] = loadings["variant"]
+    pc_loadings=pc_loadings.set_index("variant")
+    pc_loadings=pc_loadings.sort_index()
+    return pc_loadings
```

Comment:
Added intersection of sets step to assert we have the correct number of variants

Reply:
is this comment still accurate?  Could we just assert that the shape of pc_loadings is what we expect?

(And also maybe a check that it's of dtype float, since we got bitten hard by that today?

Reply:
Yes but moved that section to the function below it - we check shape in the transformation function but float check here sounds like a good idea.